### PR TITLE
Implement tests and automated known issue deployments

### PR DESCRIPTION
**Pull request by danrue**
Significant changes from the original.
- Introduced tests which run using pytest, and travis automation for tests
- Introduced travis --dry-run automation for pull requests. Build output will show changes being made to qa-reports.
- Introduced idempotency for PUTs - sync_known_issues.py will not do a PUT for existing items unless they have changed.
- Dropped .netrc support - implemented simple environment variable. Will need to be extended to support multiple squad endpoints in the future.
- Simplified the structure of the yaml files (dropping fields in 'environments')

### CODE REVIEW

File: sync_known_issues.py [hunk 1]
```diff
@@ -144,51 +141,47 @@ def __init__(self, config, squad_project, strict=False):
...
-                    if strict:
-                        raise SquadKnownIssueException(
-                            "Unknown architecture: %s" % item['architecture'])
+            if item in squad_project.environments:
+                self.environments.add(item)
+            else:
+                raise SquadKnownIssueException(
+                    "Incorrect environment: %s" % item['slug'])
+
+    def __repr__(self):
```
Comment (mwasilew) [inline sync_known_issues.py]: wouldn't it be better to use yaml for object serialization?

File: sync_known_issues.py [hunk 2]
```diff
@@ -284,35 +276,35 @@ def main():
...
+                'test_name': known_issue.test_name,
+                'url': known_issue.url,
+                'notes': known_issue.notes,
+                'active': known_issue.active,
+                'intermittent': known_issue.intermittent,
+                'environment': [item['url'] for item in affected_environments]
+            }
+
+            if api_known_issue is None:
+                print("Adding issue:")
```
Comment (mwasilew) [inline sync_known_issues.py]: use logging.info ?

File: sync_known_issues.py [hunk 3]
```diff
@@ -295,7 +323,11 @@ def main():
                         'knownissues',
                         known_issue_api_object
                     )
-            else:
+            else: # update case
+                if issues_equal(api_known_issue, known_issue_api_object):
+                    print("No changes to '{}'".format(api_known_issue['test_name']))
```
Comment (mwasilew) [inline sync_known_issues.py]: I'm OK with this concept but I don't think it's worth the effort.

File: sync_known_issues.py [hunk 4]
```diff
@@ -196,6 +198,32 @@ def parse_files(config_files):
...
+    y = b.copy()
+
+    # Copy and remove 'id' for purpose of comparison
+    # Remove 'id' field
+    if 'id' in x: del x['id']
+    if 'id' in y: del y['id']
+
+    # Sort lists for purpose of comparison
+    x['environments'].sort()
+    y['environments'].sort()
```
Comment (mwasilew) [inline sync_known_issues.py]: I'm a bit confused on how it works. Locally environments are in form of 'some_env_name'. Remotely SQUAD presents environments as 'https://squad.tld/api/environmnets/ID'. How do these compare without resolving the url to the object?

File: ltp-production.yaml [hunk 5]
```diff
@@ -1,467 +1,483 @@
...
+      LKFT: LTP: open posix: clock_settime_8-1 intermittent failures on all
+      device
+    projects: *projects_all
+    test_name: ltp-open-posix-tests/clock_settime_8-1
+    url: https://bugs.linaro.org/show_bug.cgi?id=3965
     active: true
-    intermittent: false
+    intermittent: true
+
+#  - environments:
```
Comment (mwasilew) [inline ltp-production.yaml]: delete commented part? If it's not used carrying it makes little sense to me.

File: sync_known_issues.py (hunk 1 above)
Comment (danrue) [reply sync_known_issues.py]: Sure, that's a good idea.

File: sync_known_issues.py (hunk 2 above)
Comment (danrue) [reply sync_known_issues.py]: I'll try and see how it's formatted. Sometimes print() is nice for mandatory things (as in, actual purposeful output).

File: ltp-production.yaml (hunk 5 above)
Comment (danrue) [reply ltp-production.yaml]: I wanted to preserve the work you did to bring over the content from the skiplists. I moved the other files into unused/, but the data will be used in the next couple weeks as we fill out all of the known issues.

In other words, it's temporary and useful for the moment.

File: sync_known_issues.py (hunk 1 above)
Comment (danrue) [reply sync_known_issues.py]: Fixed in https://github.com/danrue/qa-reports-known-issues/commit/aceefbae6697d228e03dc213ed2d5c158ff294c1 (i guess github is having some delay in updating the pull request, according to their status page)

**Pull request merged**